src/db_options.rs

```diff
@@ -1202,7 +1201,7 @@ impl Options {
 
         unsafe {
             let cf = ffi::rocksdb_compactionfilter_create(
-                mem::transmute(cb),
+                Box::into_raw(cb) as *mut c_void,
```

Comment:
```suggestion
                Box::into_raw(cb).cast::<c_void>(),
```

Isn't it safer it such way ?

Reply:
Didn't know this one. It's safer, because it can't accidentally go from `const *` to `mut *`. Will update the PR accordingly.